pwndbg/dbg/__init__.py

```diff
@@ -251,6 +254,39 @@ def remove(self) -> None:
         raise NotImplementedError()
 
 
+class EventType(Enum):
+    """
+    Events that can be listened for and reacted to in a debugger.
+
+    The events types listed here are defined as follows:
+        - `START`: This event is fired some time between the creation of or
+          attachment to the process to be debugged, and the start of its
+          execution.
+        - `STOP`: This event is fired after execution of the process has been
+          suspended, but before control is returned to the user for interactive
+          debugging.
+        - `EXIT`: This event is fired after the process being debugged has been
+          detached from or has finished executing.
```

Comment:
Same here: 100%  sure about detached?

Reply:
The GDB docs explicitly [mention the detach case for the exit event](https://sourceware.org/gdb/current/onlinedocs/gdb.html/Events-In-Python.html). Additionally, since we know from the previous comment that `start` does get fired on `attach`, Pwndbg already depends on the `exit` event being fired on detach to work correctly with attach/detach. Just in case, I've also confirmed experimentally that `exit` does get fired on `detach`.

As for LLDB, it's the same as the previous comment, we'll be driving the events ourselves, so we can pretty easily guarantee it gets fired on detach.